browser/caliban.py

```diff
@@ -539,210 +417,259 @@ def action_threshold(self, y1, x1, y2, x2, frame, label):
+            self.metadata.cell_info[self.feature][add_label]['frames'] = updated_frames
         # cell does not exist anywhere in npz:
         except KeyError:
-            self.file.cell_info[self.feature][add_label] = {
+            self.metadata.cell_info[self.feature][add_label] = {
                 'label': str(add_label),
                 'frames': [frame],
                 'slices': ''
             }
-            self.file.cell_ids[self.feature] = np.append(self.file.cell_ids[self.feature],
-                                                         add_label)
+            self.metadata.cell_ids[self.feature] = np.append(self.metadata.cell_ids[self.feature],
+                                                             add_label)
 
         # if adding cell, frames and info have necessarily changed
-        self._y_changed = self.info_changed = True
+        self.y_changed = self.info_changed = True
 
     def del_cell_info(self, del_label, frame):
         """Remove a cell from the npz"""
         # remove cell from frame
-        old_frames = self.file.cell_info[self.feature][del_label]['frames']
+        old_frames = self.metadata.cell_info[self.feature][del_label]['frames']
         updated_frames = np.delete(old_frames, np.where(old_frames == np.int64(frame))).tolist()
-        self.file.cell_info[self.feature][del_label]['frames'] = updated_frames
+        self.metadata.cell_info[self.feature][del_label]['frames'] = updated_frames
 
         # if that was the last frame, delete the entry for that cell
-        if self.file.cell_info[self.feature][del_label]['frames'] == []:
-            del self.file.cell_info[self.feature][del_label]
+        if self.metadata.cell_info[self.feature][del_label]['frames'] == []:
+            del self.metadata.cell_info[self.feature][del_label]
 
             # also remove from list of cell_ids
-            ids = self.file.cell_ids[self.feature]
-            self.file.cell_ids[self.feature] = np.delete(ids, np.where(ids == np.int64(del_label)))
+            ids = self.metadata.cell_ids[self.feature]
+            self.metadata.cell_ids[self.feature] = np.delete(ids,
+                                                             np.where(ids == np.int64(del_label)))
 
         # if deleting cell, frames and info have necessarily changed
-        self._y_changed = self.info_changed = True
+        self.y_changed = self.info_changed = True
 
     def create_cell_info(self, feature):
         """Make or remake the entire cell info dict"""
-        self.file.create_cell_info(feature)
+        self.metadata.create_cell_info(feature)
         self.info_changed = True
 
 
 class TrackEdit(BaseEdit):
-    def __init__(self, file_, output_bucket):
-        super(TrackEdit, self).__init__(file_, output_bucket)
+    def __init__(self, project):
+        super(TrackEdit, self).__init__(project)
 
-        self.scale_factor = 2
+        # TODO: do we need this?
+        # self.scale_factor = 2
```

Comment:
we do need this for now but it will be removed soon (once the frontend makes use of `CanvasState` for trk files--in progress)